ledger/simulation/tracer.go

```diff
@@ -284,10 +284,20 @@ func (tracer *evalTracer) BeforeOpcode(cx *logic.EvalContext) {
 		}
 		*txnTrace.programTraceRef = append(*txnTrace.programTraceRef, tracer.makeOpcodeTraceUnit(cx))
 
+		latestOpcodeTraceUnit := &(*txnTrace.programTraceRef)[len(*txnTrace.programTraceRef)-1]
 		if tracer.result.ReturnStackChange() {
-			latestOpcodeTraceUnit := &(*txnTrace.programTraceRef)[len(*txnTrace.programTraceRef)-1]
 			latestOpcodeTraceUnit.computeStackValueDeletions(cx, tracer)
 		}
+
+		if tracer.result.ReturnScratchChange() {
+			scratchSlotID, _, newValue, isStoreAlike := cx.CurrentScratchChange()
```

Comment:
seems like we indeed do not need to old value

Reply:
right, I was just thinking of the backwards support, but it seems not necessary to return old value now.